sdk/src/main/java/io/dapr/client/AbstractDaprClient.java

```diff
@@ -509,6 +510,24 @@ public Mono<Void> saveState(String storeName, String key, String etag, Object va
     return this.saveBulkState(storeName, Collections.singletonList(state));
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Mono<Void> saveState(String storeName, String key, String etag, Object value, Map<String, String> meta,
+                              StateOptions options) {
```

Comment:
It is dangerous to modify the incoming method parameter. I would rather make a copy and add the content type if `value` is not null.

Reply:
done.. thanks for the suggestion @artur-ciocanu